Consequences/Results/Result.cs

```
@@ -14,7 +14,6 @@ public ResultItem Fetch(string name)
     for (int i = 0; i < ResultItems.Length; i++)
       if (ResultItems[i].ResultName == name)
         return ResultItems[i];
-    // return empty ResultItem if not found
-    return new ResultItem { ResultName = name };
+    throw new ArgumentException($"{name} not found in result");
```

I appreciate this. We can relax this explosion later if we want, but defaulting to exploding on the error rather than hiding it is probably good practice.